src/test/java/com/easypost/ShipmentTest.java

```diff
@@ -0,0 +1,234 @@
+        Shipment retrievedShipment = Shipment.retrieve(shipment.getId());
+
+        assertTrue(retrievedShipment instanceof Shipment);
+        assertTrue(retrievedShipment.getId().startsWith("shp_"));
+        assertEquals(shipment.getId(), retrievedShipment.getId());
+    }
+
+    /**
+     * Test retrieving all shipments.
+     *
+     * @throws EasyPostException when the request fails.
+     */
+    @Test
+    public void testAll() throws EasyPostException {
+        Map<String, Object> params = new HashMap();
+
+        params.put("page_size", Fixture.pageSize());
+
+        ShipmentCollection shipments = Shipment.all(params);
+
+        assertNotNull(shipments.getHasMore());
+        assertTrue(shipments.getShipments().size() <=5);
+        for(Shipment shipment: shipments.getShipments()) {
+            assertTrue(shipment instanceof Shipment);
+        }
+    }
+
+    /**
+     * Test buying a shipment.
+     *
+     * @throws EasyPostException when the request fails.
+     */
+    @Test
+    public void testBuy() throws EasyPostException {
+        Shipment shipment = Shipment.create(Fixture.basicShipment());
+
+        Shipment boughtShipment = shipment.buy(shipment.lowestRate());
+
+        assertTrue(boughtShipment instanceof Shipment);
+        assertTrue(boughtShipment.getId().startsWith("shp_"));
+        assertNotNull(boughtShipment.getPostageLabel());;
+    }
+
+    /**
+     * Test generating a new rate for a shipment.
+     *
+     * @throws EasyPostException when the request fails.
+     */
+    @Test
+    public void testRegenerateRates() throws EasyPostException {
+        Shipment shipmentWithNewRates = shipment.newRates();
+
+        List<Rate> rates = shipmentWithNewRates.getRates();
+
+        assertTrue(rates instanceof List);
+        for(Rate rate: rates) {
+            assertTrue(rate instanceof Rate);
+        }
+    }
+
+    /**
+     * Test converting a label for a shipment.
+     *
+     * @throws EasyPostException when the request fails.
+     */
+    @Test
+    public void testConvertLabel() throws EasyPostException {
+        Map<String, Object> params = new HashMap<>();
+
+        params.put("file_format", "ZPL");
+
+        Shipment shipment = Shipment.create(Fixture.oneCallBuyShipment());
+        
+        Shipment shipmentWithLabel = shipment.label(params);
+
+        assertTrue(shipmentWithLabel instanceof Shipment);
+        assertTrue(shipmentWithLabel.getId().startsWith("shp_"));
+        assertNotNull(shipmentWithLabel.getPostageLabel().getLabelZplUrl());
+    }
+
+    /**
+     * Test adding insurace to a shipment.
+     *
+     * @throws EasyPostException when the request fails.
+     */
+    @Test
+    public void testInsure() throws EasyPostException {
+        Map<String, Object> shipmentData = Fixture.oneCallBuyShipment();
+        Map<String, Object> insuranceData = new HashMap<>();
+
+        insuranceData.put("amount", "100");
+        shipmentData.put("insurance", 0);
```

Comment:
Please add the following inline comment here: `Set to 0 so USPS doesn't insure this automatically and we can insure the shipment manually`

This explains they **why** we are setting it to 0 which is super important to call out.